execution/gethexec/sequencer.go

```diff
@@ -909,61 +910,167 @@ func (s *Sequencer) handleInactive(ctx context.Context, queueItems []txQueueItem
+	return true
+}
+
+func (s *FullSequencingHooks) SequencedTx(txId int) (*types.Transaction, error) {
 	// This is not supposed to happen, if so we have a bug
 	if txId > s.sequencedQueueItemsCount {
-		return nil, fmt.Errorf("transaction queried for was not scheduled by the fullSequencingHooks. txId: %d, sequencedCount: %d", txId, s.sequencedQueueItemsCount)
+		return nil, fmt.Errorf("transaction queried for was not scheduled by the FullSequencingHooks. txId: %d, sequencedCount: %d", txId, s.sequencedQueueItemsCount)
 	}
 	return s.queueItems[txId].tx, nil
 }
 
-func (s *Sequencer) makeSequencingHooks(items []txQueueItem) *fullSequencingHooks {
-	res := &fullSequencingHooks{
-		arbos.SequencingHooks{
-			PreTxFilter:             s.preTxFilter,
-			PostTxFilter:            s.postTxFilter,
-			DiscardInvalidTxsEarly:  true,
-			TxErrors:                []error{},
-			ConditionalOptionsForTx: nil,
-		},
-		items,
-		0,
-		0,
-		s.config().MaxTxDataSize,
+func (s *FullSequencingHooks) PreTxFilter(config *params.ChainConfig, header *types.Header, db *state.StateDB, a *arbosState.ArbosState, transaction *types.Transaction, options *arbitrum_types.ConditionalOptions, address common.Address, info *arbos.L1Info) error {
+	if s.preTxFilter != nil {
+		return s.preTxFilter(config, header, db, a, transaction, options, address, info)
+	}
+	return nil
+}
+
+func (s *FullSequencingHooks) PostTxFilter(header *types.Header, db *state.StateDB, a *arbosState.ArbosState, transaction *types.Transaction, address common.Address, u uint64, result *core.ExecutionResult) error {
+	if s.postTxFilter != nil {
+		return s.postTxFilter(header, db, a, transaction, address, u, result)
+	}
+	return nil
+}
+
+func (s *FullSequencingHooks) BlockFilter(header *types.Header, db *state.StateDB, transactions types.Transactions, receipts types.Receipts) error {
+	if s.blockFilter != nil {
+		return s.blockFilter(header, db, transactions, receipts)
+	}
+	return nil
+}
+
+func MakeSequencingHooks(
+	items []txQueueItem,
+	maxSequencedTxsSize int,
+	preTxFilter func(*params.ChainConfig, *types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, *arbitrum_types.ConditionalOptions, common.Address, *arbos.L1Info) error,
+	postTxFilter func(*types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, common.Address, uint64, *core.ExecutionResult) error,
+	blockFilter func(*types.Header, *state.StateDB, types.Transactions, types.Receipts) error,
+) *FullSequencingHooks {
+	res := &FullSequencingHooks{
+		queueItems:               items,
+		sequencedQueueItemsCount: 0,
+		sequencedTxsSizeSoFar:    0,
+		maxSequencedTxsSize:      maxSequencedTxsSize,
+		preTxFilter:              preTxFilter,
+		postTxFilter:             postTxFilter,
+		blockFilter:              blockFilter,
 	}
-	res.SequencingHooks.NextTxToSequence = res.GetNextTx
-	res.SequencingHooks.SequencedTx = res.GetScheduledTx
 	return res
 }
 
+func MakeZeroTxSizeSequencingHooks(
```

Comment:
Took me some time to figure out what you're doing here.
I like it. It's a good solution. But maybe either add a comment that explains it or add a small comment.